cells/lib/ops/revise.nix

```diff
@@ -0,0 +1,65 @@
+{
+  inputs,
+  cell,
+}: let
+  inherit (inputs) nixpkgs;
+  l = nixpkgs.lib // builtins;
+in
+  /*
+  For use with `revisePackage` and `reviseOCI` to build containers in a mono-repo style
+  environment where the source code is contained in the same repository as the std code,
+  specifically so that one may detect meaningful changes to the image via its tag in the
+  special case where the package's output includes the revision of the source code (e.g. for
+  displaying the version to the user).
+
+  Without special processing, this kind of package would cause the OCI image tag to change
+  on each new revision whether the actual contents of the image changed or not. Combined
+  with `std.incl`, one may have a very strong indicator for when the contents of the image
+  actually includes meaningful changes which avoids flooding the remote registry with superflous
+  copies.
+
+  Args:
+  op: Optional function which takes the package as an argument.
+  pkg: The package you wish to revise.
+  fn: Optional functor with a reference to `pkg` if needed by `op`.
+
+  Returns:
+  The package with a clone of itself in the passthru where the expected revision is set to
+  "not-a-commit" for later use by `reviseOCI`.
```

Comment:
```
and `revisePackage`
```

Reply:
well the order as far as the user is concerned is `revisePackage` -> `revise` (on the operable) -> `reviseOCI`, so its speaking to that sequence. The use of revise in `revisePackage` is strictly an implementation detail

Reply:
Does this mean `revidePackage` can't be used standalone for this purpose?